### PR TITLE
8294 - Handle invalid table name

### PR DESCRIPTION
**Pull request by JohJohan**
See https://github.com/doctrine/orm/issues/8294

### CODE REVIEW

**Comment by Ahummeling** [comment]
I think the problem here (programmer error) is solved in the wrong place. The annotations reader is responsible for reading the annotations. If the syntax is wrong (a keyword is missing), and if the reader is able to detect this as an error, then the reader should throw an exception to inform the programmer he is making a mistake. 

The problem I have with this solution is that checking whether the `$tabe['name']` is a string is that it still doesn't guarantee that it's actually the name, and not some other property of the table annotation that also happens to be a string. 

In short, I think the issue lies with incorrect mapping of annotation.

**Comment by JohJohan** [comment]
@Ahummeling i can see where your comming from, maybe a doctrine maintainer can give a hint where the fix needs to be placed

**Comment by greg0ire** [comment]
@Ahummeling already gave the correct hint IMO:

> The annotations reader

**Comment by JohJohan** [comment]
@greg0ire i create a branch on doctrine/annotations see https://github.com/doctrine/annotations/pull/447 it looks like it is not a problem anymore for tag 1.13.x because we get a valid name see my test. Let me know what you think?

**Comment by greg0ire** [comment]
@JohJohan sorry I didn't double check @Ahummeling 's hint more . The right place to do this is IMO inside the annotation _driver_ (so inside the ORM). The annotation reader doesn't know about ORM-specific annotations at all (nor should it).

https://github.com/doctrine/orm/blob/2.13.x/lib/Doctrine/ORM/Mapping/Driver/AnnotationDriver.php

**Comment by JohJohan** [comment]
@greg0ire i made changes in the [AnnotationDriver](https://github.com/doctrine/orm/blob/2.13.x/lib/Doctrine/ORM/Mapping/Driver/AnnotationDriver.php) maybe you can do a CR?
